keps/sig-node/2000-graceful-node-shutdown/README.md

```diff
@@ -0,0 +1,843 @@
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Kubelet should be aware of node shutdown and trigger graceful shutdown of pods
+during a machine shutdown.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Users and cluster administrators expect that pods will adhere to expected pod
+lifecycle including pod termination. Currently, when a node shuts down, pods do
+not follow the expected pod termination lifecycle and are not terminated
+gracefully which can cause issues for some workloads. This KEP aims to address
+this problem by making the kubelet aware of the underlying node shutdown.
+Kubelet will propagate this signal to pods ensuring they can shutdown as
+gracefully as possible.
+
+### Goals
```

Comment:
may be worth mentioning for scoping - does this KEP tries to address the expedited re-scheduling of pods or notification of API server that node is about to shutdown?

Reply:
btw, there is another thing about shutdown: https://github.com/kubernetes/kubernetes/issues/58635 . That is resulting in this KEP: https://github.com/kubernetes/enhancements/issues/551. Just for the reference on other issues in this area. Just found out about it

Reply:
Thanks Sergey for pointing that out. Doesn't look like there is any activity on that KEP from October 2019, so will resolve this for now.